q2_amr/card/database.py

```diff
@@ -16,30 +17,72 @@
 
 
 def fetch_card_db() -> (CARDDatabaseDirectoryFormat, CARDKmerDatabaseDirectoryFormat):
-    # Fetch CARD and WildCARD data from CARD website
-    try:
-        response_card = requests.get(
-            "https://card.mcmaster.ca/latest/data", stream=True
-        )
-        response_wildcard = requests.get(
-            "https://card.mcmaster.ca/latest/variants", stream=True
-        )
-    except requests.ConnectionError as e:
-        raise requests.ConnectionError("Network connectivity problems.") from e
-
-    # Create temporary directory for WildCARD data
     with tempfile.TemporaryDirectory() as tmp_dir:
+        try:
+            response_card = requests.get(
+                "https://card.mcmaster.ca/latest/data", stream=True
+            )
+            response_wildcard = requests.get(
+                "https://card.mcmaster.ca/latest/variants", stream=True
+            )
+
+            # Get content length to calculate progress bar length
+            tot_size_card = int(response_card.headers.get("content-length", 0))
+            tot_size_wildcard = int(response_wildcard.headers.get("content-length", 0))
+
+            # Paths for database tar archives
+            card_path = os.path.join(tmp_dir, "card_tar")
+            wildcard_path = os.path.join(tmp_dir, "wildcard_tar")
+
+            # Initialize CARD progress bar and download database
+            progress_bar_card = tqdm(
+                total=tot_size_card,
+                unit="B",
+                unit_scale=True,
+                desc="Downloading CARD database",
+            )
+
+            with open(card_path, "wb") as file:
+                for chunk in response_card.iter_content(chunk_size=8192):
+                    file.write(chunk) if chunk else False
+                    if tot_size_card > 0:
+                        progress_bar_card.update(len(chunk))
+                progress_bar_card.close() if tot_size_card > 0 else False
+
+            # Initialize WildCARD progress bar and download database
+            progress_bar_wildcard = tqdm(
+                total=tot_size_wildcard,
+                unit="B",
+                unit_scale=True,
+                desc="Downloading WildCARD database",
+            )
+
+            with open(wildcard_path, "wb") as file:
+                for chunk in response_wildcard.iter_content(chunk_size=8192):
+                    file.write(chunk) if chunk else False
+                    if tot_size_wildcard > 0:
+                        progress_bar_wildcard.update(len(chunk))
+                progress_bar_wildcard.close() if tot_size_wildcard > 0 else False
```

Comment:
You are running pretty much the same thing _twice_. Please extract a method and refactor this function accordingly to simplify.